code/game/machinery/doors/airlock.dm

```diff
@@ -117,6 +118,27 @@
 		max_integrity = normal_integrity
 	if(damage_deflection == AIRLOCK_DAMAGE_DEFLECTION_N && security_level > AIRLOCK_SECURITY_METAL)
 		damage_deflection = AIRLOCK_DAMAGE_DEFLECTION_R
+	if(abandoned)
+		var/outcome = rand(1,100)
+		switch(outcome)
+			if(1 to 9)
+				var/turf/here = get_turf(src)
+				for(var/turf/closed/T in range(2, src))
```

Comment:
How would this look in practice? Would this interfere with objects or loot spawners placed near the airlock?

Reply:
I mean it should search for a nearby wall and basically delete the airlock and place that wall type there instead.

I figured it would be a more dynamic effect if it ends up being used in areas with non-standard walls. I'm not sure how this could interfere with loot spawns since at the end of the day it just creates a wall and deletes an airlock - nothing outside of the airlock's turf gets effected.